features/newts-repository-converter/src/main/java/org/opennms/features/newts/converter/NewtsConverter.java

```diff
@@ -0,0 +1,613 @@
+        if (storeByGroup) {
+            this.processStoreByGroupResources(this.rrdDir.resolve("snmp"));
+        } else {
+            this.processStoreByMetricResources(this.rrdDir.resolve("snmp"));
+        }
+    }
+
+    private void processStoreByGroupResources(final Path path) {
+        try {
+            // Find and process all resource folders containing a 'ds.properties' file
+            Files.walk(path)
+                 .filter(p -> p.endsWith("ds.properties"))
+                 .forEach(p -> processStoreByGroupResource(p.getParent()));
+
+        } catch (Exception e) {
+            throw NewtsConverterError.create(e, "Error while reading RRD files: {}", e.getMessage());
+        }
+    }
+
+    private void processStoreByGroupResource(final Path path) {
+        // Load the 'ds.properties' for the current path
+        final Properties ds = new Properties();
+        try (final BufferedReader r = Files.newBufferedReader(path.resolve("ds.properties"))) {
+            ds.load(r);
+
+        } catch (final IOException e) {
+            throw NewtsConverterError.create("No group information found - please verify storeByGroup settings");
+        }
+
+        // Get all groups declared in the ds.properties and process the RRD files
+        Sets.newHashSet(Iterables.transform(ds.values(), Object::toString))
+            .forEach(group -> this.executor.execute(() -> processResource(path,
+                                                                          group,
+                                                                          group)));
+    }
+
+    private void processStoreByMetricResources(final Path path) {
+        try {
+            // Find an process all '.meta' files and the according RRD files
+            Files.walk(path)
+                 .filter(p -> p.getFileName().toString().endsWith(".meta"))
+                 .forEach(p -> this.processStoreByMetricResource(p));
+
+        } catch (Exception e) {
+            throw NewtsConverterError.create(e, "Error while reading RRD files: {}", e.getMessage());
+        }
+    }
+
+    private void processStoreByMetricResource(final Path metaPath) {
+        // Use the containing directory as resource path
+        final Path path = metaPath.getParent();
+
+        // Extract the metric name from the file name
+        final String metric = FilenameUtils.removeExtension(metaPath.getFileName().toString());
+
+        // Load the '.meta' file to get the group name
+        final Properties meta = new Properties();
+        try (final BufferedReader r = Files.newBufferedReader(path.resolve(String.format("%s.meta", metric)))) {
+            meta.load(r);
+
+        } catch (final IOException e) {
+            throw Throwables.propagate(e);
+        }
+
+        final String group = meta.getProperty("GROUP");
+        if (group == null) {
+            throw NewtsConverterError.create("No group information found - please verify storeByGroup settings");
```

Comment:
It's possible that one or more archives do not have the GROUP defined in their meta files i.e. a some metrics were collected for an interface that no longer exists.

I think we should log the error and indicate the name of the file in question, but continue to process the remaining archives when this happens.